src/components/weather/CityForecast.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { fetchCityForecast } from "../../services/weatherService";
+import {
+  ForecastContainer,
+  ForecastHeader,
+  ForecastItem,
+  DayName,
+  TemperatureWrapper,
+  ResponsiveForecastList,
+  WeatherImage,
+  DayTemp,
+} from "./styled";
+
+// eslint-disable-next-line @typescript-eslint/no-redeclare
+interface ForecastItem {
+  dayName: string;
+  minTemp: number;
+  maxTemp: number;
+  weatherDescription: string;
+  weatherIcon?: string;
+}
+
+const CityForecast: React.FC = () => {
+  const { cityName } = useParams<{ cityName?: string }>();
+  const [forecastData, setForecastData] = useState<ForecastItem[]>([]);
+  const OPEN_WEATHER_URL = "https://openweathermap.org/img/w/";
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        if (cityName) {
+          const data = await fetchCityForecast(cityName);
+          setForecastData(data);
+        }
+      } catch (error) {
+        console.error("Error fetching city forecast:", error);
+      }
+    };
+
+    fetchData();
+  }, [cityName]);
+
+  return (
+    <ForecastContainer>
+      <ForecastHeader>{cityName} 5-Day Forecast</ForecastHeader>
+      <ResponsiveForecastList>
+        {forecastData.map((item, index) => (
+          <ForecastItem key={index}>
+            <DayName>{item.dayName}</DayName>
+            {item.weatherIcon && (
+              <WeatherImage
+                src={`${OPEN_WEATHER_URL}${item.weatherIcon}.png`}
```

Comment:
don't pass src as props, pass just weatherIcon and then define route inside `WeatherImage`